src/qibo/backends/numpy.py

```diff
@@ -471,14 +476,27 @@ def execute_circuit_repeated(self, circuit, initial_state=None, nshots=None):
             else:
                 results.append(state)
 
+            if not circuit.density_matrix:
+                probabilities += result.probabilities()
+
+        if not circuit.density_matrix:
+            probabilities /= nshots
+
         if circuit.measurements:
             final_result = CircuitResult(self, circuit, state, nshots)
             final_result._samples = self.aggregate_shots(results)
+            if not circuit.density_matrix:
+                final_result.probabilities = (
+                    lambda: self._override_probabilities_repeated_execution(
+                        probabilities
+                    )
+                )
```

Comment:
> It isn't a pretty fix, but it does the job.

This really seems quite nasty :joy: I hope @vodovozovaliza does not take it as an example of good coding practices.

Reply:
hahahaha from the start I said it wasn't pretty